src/test/java/org/util/string/StringUtilTest.java

```diff
@@ -1,70 +1,191 @@
+        assertEquals('q', actual);
+    }
+
+    @Test
+    public void testAppendStringBufferData() {
+        StringUtil input = new StringUtil("qwerty");
+        StringUtil actual = input.append(new StringBuffer("asdfg"));
+        assertEquals("qwertyasdfg", actual.toString());
+    }
+
+    @Test
+    public void testAppendEmptyStringBufferData() {
+        StringUtil input = new StringUtil("qwerty");
+        StringUtil actual = input.append(new StringBuffer(""));
+        assertEquals("qwerty", actual.toString());
+    }
+
+    @Test
+    public void testPrependStringBufferData() {
+        StringUtil input = new StringUtil("qwerty");
+        StringUtil actual = input.prepend(new StringBuffer("asdfg"));
+        assertEquals("asdfgqwerty", actual.toString());
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"this", "is ", "ball", "implementation",})
+    public void testReverseWhenStringUtilHaveDifferentSize(String argument) {
+        StringUtil input = new StringUtil(argument);
+        StringUtil actual = input.reverse();
+        String expected = new String(new StringBuffer(argument).reverse());
+        assertEquals(expected, actual.toString());
+    }
+
+    @Test
+    public void testReverseWhenStringUtilIsEmpty() {
+        StringUtil input = new StringUtil("");
+        StringUtil actual = input.reverse();
+        assertEquals("", actual.toString());
+    }
+
+    @ParameterizedTest
+    @EnumSource(Palindrome.class)
+    public void testReverseWhenInStringUtilOneElement() {
+        List<StringUtil> stringArray = new ArrayList<>();
+        stringArray.add(new StringUtil(new char[]{'k'}));
+        stringArray.add(new StringUtil(new char[]{'r'}));
+        stringArray.add(new StringUtil(new char[]{'d'}));
+
+        assertAll(() -> assertEquals(Palindrome.k.toString(), stringArray.get(0).reverse().toString()),
+                () -> assertEquals(Palindrome.r.toString(), stringArray.get(1).reverse().toString()),
+                () -> assertEquals(Palindrome.d.toString(), stringArray.get(2).reverse().toString())
+        );
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"kayak", "rotator", "dad"})
+    public void testReverseWhenStringUtilIsPalindrome(String palyndrom) {
+        StringUtil actual = new StringUtil(palyndrom);
+        assertEquals(palyndrom, actual.reverse().toString());
+    }
+
+    @Test
+    public void testSetWhenStringUtilIsEmpty() {
+        StringUtil input = new StringUtil("");
+        StringUtil actual = input.set(0, 'a');
+        assertEquals("a", actual.toString());
+    }
+
+    @Test
+    public void testSetElementsInDifferentPlaces() {
+        StringUtil input = new StringUtil("kvadrat");
+
+        assertAll("testing set first, last and central elements",
+                () -> assertEquals("svadrat", input.set(0, 's').toString()),
+                () -> assertEquals("kvadrad", input.set(input.size() - 1, 'd').toString()),
```

Comment:
instead of calculations you should explicitly set indexes, - use `5` instead of `size() - 2`